storage/client_raft_test.go

```diff
@@ -1598,3 +1598,46 @@ func TestCheckInconsistent(t *testing.T) {
 		t.Fatal("didn't receive notification from VerifyChecksum() that should have panicked")
 	}
 }
+
+func TestTransferRaftLeadership(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const numStores = 3
+	mtc := startMultiTestContext(t, numStores)
+	defer mtc.Stop()
+	// Setup replication of ramge 1 on store 0 to stores 1 and 2.
+	mtc.replicateRange(1, 1, 2)
+
+	// Write something to the DB to make range leadership and raft leadership to be located in stores[0].
+	pArgs := putArgs([]byte("a"), []byte("b"))
+	if _, err := client.SendWrapped(rg1(mtc.stores[0]), nil, &pArgs); err != nil {
```

Comment:
This could be flaky if a background process on another store gets there first. What if you assume leadership before replicating out?